File: zbus-api-csharp/zbus/Proto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using zbus.Remoting;
using fastJSON;
namespace zbus
{
    public enum MessageMode
    {
        MQ = 1<<0,
        PubSub = 1<<1,
        Temp = 1<<2,
    }

    public class Proto
    {
        public static readonly string Produce     = "produce";    //生产消息
        public static readonly string Consume     = "consume";    //消费消息 
        public static readonly string Request     = "request";    //请求等待应答消息  
        public static readonly string Heartbeat   = "heartbeat";  //心跳消息
        public static readonly string Admin       = "admin";      //管理类消息
        public static readonly string CreateMQ    = "create_mq";      
        public static readonly string TrackReport = "track_report";
        public static readonly string TrackSub    = "track_sub";
        public static readonly string TrackPub    = "track_pub";


        public static Message BuildSubCommandMessage(string cmd, string subcmd, IDictionary<string, string> args)
        {
            Message msg = new Message();
            msg.Command = cmd;
            msg.SubCommand = subcmd;
            msg.SetJsonBody(JSON.Instance.ToJSON(args));
            return msg;
        }
    }
}

File: src/main/java/org/zstacks/zbus/protocol/Proto.java
package org.zstacks.zbus.protocol;

import java.util.Map;

import org.zstacks.znet.Message;

import com.alibaba.fastjson.JSON;

public class Proto {
	public static final String Heartbeat = Message.HEARTBEAT; //心跳消息 
	public static final String Produce   = "produce";         //生产消息
	public static final String Consume   = "consume";         //消费消息  
	public static final String Request   = "request";         //请求等待应答消息 
	public static final String Admin     = "admin";           //管理类消息  
	//管理类命令二级子命令 
	public static final String AdminCreateMQ    = "create_mq";  
	public static final String AdminQueryMQ     = "query_mq";  
	
	//TrackServer命令
	public static final String TrackReport      = "track_report"; 
	public static final String TrackSub         = "track_sub";  
	public static final String TrackPub         = "track_pub"; 
	//5.2.0 added
	public static final String TrackQuery       = "track_query"; 
	
	
	public static Message buildSubCommandMessage(String cmd, String subCmd, Map<String, String> params){
    	Message msg = new Message();
    	msg.setCommand(cmd); 
    	msg.setSubCommand(subCmd); 
    	if(params != null){
    		msg.setJsonBody(JSON.toJSONBytes(params));
    	}
    	return msg;
    }
	 
}
